grammarinator/generate.py

```diff
@@ -286,10 +301,14 @@ def restricted_float(value):
                    transformers=args.transformer, serializer=args.serializer,
                    cleanup=False, encoding=args.encoding) as generator:
         if args.jobs > 1:
-            with Pool(args.jobs) as pool:
-                pool.map(generator, range(args.n))
+            with Manager() if not args.out else nullcontext() as manager:
+                if not args.out:
+                    generator = partial(generator, lock=manager.Lock())
```

Comment:
pylint [complains](https://github.com/renatahodovan/grammarinator/runs/2838943063?check_suite_focus=true) about this line, but it seems that it's a [known pylint issue](https://github.com/PyCQA/pylint/issues/3313), so we can just ignore it. This should work, but you can validate it by running tox locally:

```py
generator = partial(generator, lock=manager.Lock())  # pylint: disable=no-member
```